woocommerce/woocommerce-ios
# Blaze: Add UI for budget setting screen

Part of #11599 


## Description

This PR adds a new screen for setting up the budget for a Blaze campaign. This includes only the UI logic and budget calculation for the screen. Fetching and displaying estimated impressions will be added in a separate PR. Unit tests will also be added later.

## Testing instructions

- Log in to a store with at least one existing product and is eligible for Blaze.
- Navigate to the Products tab and select any public product.
- On the product form, select Promote with Blaze.
- Notice that the campaign creation form is presented.
- Select the budget field, notice that a new screen for budget setting is displayed.
- Tap the estimated impression text, a new sheet should be displayed for more information on impressions.
- Tap the Edit button on the date range to change the duration and start date. The formatted duration should be updated correctly.
- Update the budget with the budget slider, the formatted amount should be updated correctly.
- Tap Update button, the details should be updated correctly on the budget field of the campaign creation form.

📝 There's still an ongoing discussion regarding formatting the budget amount on Figma - I'll update the format later after we reach a conclusion.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/48a4a3ae-4bcf-40e5-8850-6d4bf4db9de8



---
